src/components/PopupWithForm.js

```diff
@@ -27,6 +30,20 @@ export default class PopupWithForm extends Popup {
     });
   }
 
+  setLoadingState(isLoading) {
+    const submitButton = this._popupForm.querySelector(".popup__button");
+
+    if (submitButton) {
+      if (isLoading) {
+        submitButton.textContent = "Saving...";
+      } else {
+        submitButton.textContent = this._submitButtonText;
+      }
+    } else {
+      console.error("Submit button not found in the form.");
+    }
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`